apis/core/test/domain/csv-source/delete.test.ts

```diff
@@ -25,17 +22,16 @@ describe('domain/csv-sources/delete', () => {
       deleteFile: sinon.spy(),
     }
 
-    function * tableGenerator() {
-      yield 'table'
+    async function * tableGenerator() {
+      yield $rdf.namedNode('table')
     }
 
-    getLinkedTablesForSource = sinon.stub().returns(tableGenerator())
-    getTablesForMapping = sinon.stub().returns(tableGenerator())
-    tableQueries = {
-      getLinkedTablesForSource,
-      getTablesForMapping,
-      getTableForColumnMapping,
-    }
+    sinon.restore()
+    sinon.stub(TableQueries, 'getLinkedTablesForSource').returns(tableGenerator())
```

Comment:
What made you decide to stub stuff instead of injecting dependencies?

Reply:
I think it's cleaner on the code which actually uses them. What do you think?

Reply:
I'm fine with both. Mocking in this way makes the code a bit simpler, but the test a bit more brittle. The tradeoff is probably worth it.